app/views/items/_category_form.html.erb

```diff
@@ -0,0 +1,8 @@
+<p>
+  <%= form_for Category.new, html: { class: 'form-inline' } do |f| %>
```

Comment:
Presenters are an interesting pattern but you may struggle a bit with how to implement them from that blog post alone @Ru-T.

The overall issue here is that the view itself is concerned with multiple things. Items, categories, and an item creation from all appear on the same page. This craziness is 100% intentional bc it is 100% realistic that a client would request that we create UI like this. I've designed the inventory manager in this way intentionally to demonstrate that sometimes clients will request that you may not be able to implement cleanly with good OOD using the models that you have.

Well, abstraction to the rescue! Take the following class into consideration that defines an object that is less about being apart of your rails ORM and more about details this business concept of an inventory management dashboard that you could use to craft this view

```
class InventoryManagerDashboard
  attr_reader :items

  def initialize(items)
    @items = items
  end

  def new_item(attrs = {})
    @new_item ||= Item.new(attrs)
  end

  def inventory_categories
    Category.all
  end

  def save_piece_of_inventory(attr)
    new_item(attrs).save
  end
end
```

You could them use this in your inventory controller maybe?

```
...
def index
  @inventory_manager = InventoryManagerDashboard.new
end

def create
  @inventory_manager = InventoryManagerDashboard.new
  if @inventory_manager.save_piece_of_inventory(item_params)
    flash[:error] = 'error message'
    render :index
  else
    flash[:success] = 'success'
    redirect_to root_path
  end
end
...
```

We can talk more details about this later. In my humble opinion, I think that calling Item.new in your view is 100% okay but the *deeper* answer to this is to implement a higher level of abstraction since it is clear that this management system UI is core to the overall application and may require some additional extension and refactoring in the future. Whenever I see/sense/smell something like that, I go straight into creating new object. I find it makes things easier when the client walks up to you one day and says "Gee, wouldn't it be nice if we had x in the inventory manager too?"